app/services/create_edition_link_monitor.rb

```diff
@@ -0,0 +1,48 @@
+class CreateEditionLinkMonitor
+  def initialize(edition)
+    @edition = edition
+    @uris = extract_links
+  end
+
+  def perform!
+    if can_perform?
+      ActiveRecord::Base.transaction do
+        upsert_resource_monitor
+      end
+
+      true
+    end
+  end
+
+  def can_perform?
+    !failure_reason
+  end
+
+  def failure_reason
+    "This edition has no links" unless has_links?
+  end
```

Comment:
Initially I thought it'd be better not to enqueue jobs for things that don't have links, but it's nice to keep the knowledge of how to extract links in here so I guess that makes sense. 👍